code_search/src/code_search/dataflow/cli/preprocess_github_dataset.py

```diff
@@ -59,8 +58,8 @@ def preprocess_github_dataset(argv=None):
       input_records = (pipeline
         | "Read Github Dataset" >> gh_bq.ReadGithubDataset(args.project))
     token_pairs = (input_records
-      | "Transform Github Dataset" >> github_dataset.TransformGithubDataset(args.project,
-                                                                            args.target_dataset)
+      | "Transform Github Dataset" >> github_dataset.TransformGithubDataset(
```

Comment:
We don't need to fix it in this PR since its a legacy of the existing code; but in general its better not to bundle sinks and sources in a transform. This makes the transform less reusable; e.g. you can't easily change the pipeline to write to a different type of sink or to do downstream processing on the results.

So a better pattern would be for TransformGitHubDataset to return token_pairs and failed_tokens and then here we would decide where to write them.

Its possible for a Transform to have multiple outputs. However, another way do this is to replace TransformGitHubDataset with a normal python function that would take input_records as an argument and return token_pairs, failed_tokenize_table.

Reply:
sounds good. just realized the two table parameters are optional. it will skip storing them if they are not specified. in this case, i'll skip specifying them.